src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/AbstractCatalogBackupRestoreTasklet.java

```diff
@@ -0,0 +1,465 @@
+
+    protected XStreamPersister xstream;
+
+    protected XStream xp;
+    
+    private boolean isNew;
+
+    private AbstractExecutionAdapter currentJobExecution;
+
+    private boolean dryRun;
+
+    private boolean bestEffort;
+
+    public AbstractCatalogBackupRestoreTasklet(Backup backupFacade,
+            XStreamPersisterFactory xStreamPersisterFactory) {
+        this.backupFacade = backupFacade;
+        this.xStreamPersisterFactory = xStreamPersisterFactory;
+    }
+    
+    @Override
+    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) 
+            throws Exception {
+        // Accordingly to the running execution type (Backup or Restore) we
+        // need to validate resources against the official GeoServer Catalog (Backup)
+        // or the temporary one (Restore).
+        //
+        // For restore operations the order matters.
+        JobExecution jobExecution = chunkContext.getStepContext().getStepExecution()
+                .getJobExecution();
+        this.xstream = xStreamPersisterFactory.createXMLPersister();
+        if (backupFacade.getRestoreExecutions() != null
+                && !backupFacade.getRestoreExecutions().isEmpty()
+                && backupFacade.getRestoreExecutions().containsKey(jobExecution.getId())) {
+            this.currentJobExecution = backupFacade.getRestoreExecutions().get(jobExecution.getId());
+            this.catalog = backupFacade.getRestoreExecutions().get(jobExecution.getId()).getRestoreCatalog();
+            this.isNew = true;
+        } else {
+            this.currentJobExecution = backupFacade.getBackupExecutions().get(jobExecution.getId());
+            this.catalog = backupFacade.getCatalog();
+            this.xstream.setExcludeIds();
+            this.isNew = false;
+        }
+
+        Assert.notNull(catalog, "catalog must be set");
+
+        this.xstream.setCatalog(this.catalog);
+        this.xstream.setReferenceByName(true);
+        this.xp = this.xstream.getXStream();
+
+        Assert.notNull(this.xp, "xStream persister should not be NULL");
+        
+        JobParameters jobParameters = this.currentJobExecution.getJobParameters();
+        
+        this.dryRun = Boolean.parseBoolean(jobParameters.getString(Backup.PARAM_DRY_RUN_MODE, "false"));
+        this.bestEffort = Boolean.parseBoolean(jobParameters.getString(Backup.PARAM_BEST_EFFORT_MODE, "false"));
+
+        return doExecute(contribution, chunkContext, jobExecution);
+    }
+    
+    /**
+     * 
+     * @param contribution
+     * @param chunkContext
+     * @param jobExecution 
+     * @return
+     * @throws Exception
+     */
+    abstract RepeatStatus doExecute(StepContribution contribution, ChunkContext chunkContext, JobExecution jobExecution)
+            throws Exception;
+
+    /**
+     * @param targetBackupFolder
+     * @throws IOException
+     */
+    public void backupGWCSettings(Resource targetBackupFolder) {
```

Comment:
I thought it could be useful for other tasklets, but in the end it wasn't so useful. It can also be removed I guess.